src/cache.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+// TODO: move to its own package
+
+import fs from 'fs-extra'
+import moment from 'moment'
+
+export default class {
+  static async fetch (cachePath: string, cacheDuration: number, options: any = {}): Promise<?Array<?string>> {
+    let cache
+    let cachePresent = await fs.exists(cachePath)
+    if (cachePresent) {
+      if (this._isStale(cachePath, cacheDuration)) {
```

Comment:
@RasPhilCo could you change this to so that you can share some logic on missing / stale caches?
```
if (cachePresent && !this._isStale(cachePath, cacheDuration)) {
  return await fs.readJSON(cachePath)
}

const cache = await options.cacheFn()
await this._updateCache(cachePath, cache)
return cache
```